pkg/watch/registrar.go

```diff
@@ -95,6 +96,15 @@ func (r *recordKeeper) RemoveRegistrar(parentName string) error {
 func (r *recordKeeper) Update(parentName string, m vitalsByGVK) {
 	r.intentMux.Lock()
 	defer r.intentMux.Unlock()
+
+	defer func() {
```

Comment:
reporting the intent by parent could be interesting, but in order to have a 1:1 expectation between watched vs. intent, we should use the value from `len(r.Get())`, though we'll need to relocate the core `r.Get()` code into internal-only version of `r.Get()` that does not acquire a lock.

Reply:
updated to use r.get(), let me know what you think